File: pact-jvm-matchers/src/main/java/au/com/dius/pact/matchers/Matcher.java
package au.com.dius.pact.matchers;

import java.util.List;
import java.util.Map;

public abstract class Matcher {
    public abstract <Mismatch> List<Mismatch> domatch(Map<String, ?> matcherDef, String path,
                                                          Object expected, Object actual,
                                                          MismatchFactory<Mismatch> mismatchFactory);

    public static String valueOf(Object value) {
        if (value instanceof String) {
            return "'" + value + "'";
        } else if (value == null) {
            return "null";
        } else {
            return value.toString();
        }
    }
}

File: pact-jvm-matchers/src/main/scala/au/com/dius/pact/matchers/Matcher.scala
package au.com.dius.pact.matchers

trait Matcher {
    def domatch[Mismatch](matcherDef: Map[String, String], path: String, expected: Any, actual: Any,
                mismatchFactory: MismatchFactory[Mismatch]) : List[Mismatch]

    def valueOf(value: Any) = {
        value match {
            case null => "null"
            case s: String => "'" + value + "'"
            case _ => value.toString
        }
    }
}
